app/services/solidus_pay_tomorrow/client/create_order_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module SolidusPayTomorrow
+  module Client
+    class CreateOrderService < SolidusPayTomorrow::Client::BaseService
+      attr_reader :order
+
+      CREATE_ENDPOINT = 'api/application/ecommerce/orders'
+
+      # @param order [Spree::Order] Spree order
+      # @param payment_method [SolidusPayTomorrow::PaymentMethod]
+      def initialize(order:, payment_method:)
+        @order = order
+        super
+      end
+
+      def call
+        create
+      end
+
+      private
+
+      def create
+        handle_errors(HTTParty.post(uri, headers: auth_headers, body: create_body.to_json))
+      end
+
+      def uri
+        "#{api_base_url}/#{CREATE_ENDPOINT}"
+      end
+
+      def create_body
+        { orderId: order.number,
+          # TODO: We'll need to implement firstname and lastname. They are deprecated in Solidus
```

Comment:
You can use this [model](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/address/name.rb) to retrieve firstname and lastname

Reply:
Used in https://github.com/nebulab/solidus_pay_tomorrow/pull/7/commits/390014a30a052261cae08adcd7221d7641aaf79f